src/components/shared/TenantSelector.tsx

```diff
@@ -0,0 +1,96 @@
+import { Skeleton } from '@mui/material';
+import AutocompletedField from 'components/shared/toolbar/AutocompletedField';
+import { useTenantDetails } from 'context/fetcher/Tenant';
+import useGlobalSearchParams, {
+    GlobalSearchParams,
+} from 'hooks/searchParams/useGlobalSearchParams';
+import { noop } from 'lodash';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useTenantStore } from 'stores/Tenant/Store';
+import { hasLength } from 'utils/misc-utils';
+
+interface Props {
+    updateStoreState?: (value?: string) => void;
+}
+
+function TenantSelector({ updateStoreState }: Props) {
+    const intl = useIntl();
+
+    const tenants = useTenantDetails();
+    const selectedTenant = useTenantStore((state) => state.selectedTenant);
+    const setSelectedTenant = useTenantStore(
+        (state) => state.updateSelectedTenant
+    );
+
+    const handledDefault = useRef(false);
+    const defaultSelectedTenant = useGlobalSearchParams(
+        GlobalSearchParams.PREFIX
+    );
+    const [inputValue, setInputValue] = useState('');
+    const [defaultValue, setDefaultValue] = useState<string | null>(null);
+
+    const tenantNames = useMemo<string[] | null>(
+        () =>
+            tenants && tenants.length > 0
+                ? tenants.map(({ tenant }) => tenant)
+                : null,
+        [tenants]
+    );
+
+    const updateState = useCallback(
+        (value: string) => {
+            if (updateStoreState) {
+                updateStoreState(value);
+            }
```

Comment:
My gut feeling is this should _probably_ happen after the local updates. That way the local is updated and rendered correctly before anything else is updated. Very nit picky and not a big deal.

Reply:
Oh that's fair. Let's leave it then to keep things simpler.